src/map/Map.js

```diff
@@ -338,8 +340,21 @@ L.Map = L.Evented.extend({
 
 		padding = L.point(padding || [0, 0]);
 
+		if (snap <= 0) {
+			zoom = this.getZoom();
+			boundsSize = this.project(se, zoom).subtract(this.project(nw, zoom)).add(padding);
+			var scale = Math.max(boundsSize.x / size.x, boundsSize.y / size.y);
+			if (Math.log2) {
+				return zoom + Math.log2(1 / scale);	// ES6 only
+			} else {
+				return zoom + (Math.log(1 / scale) / Math.log(2));
+			}

```

Comment:
Speaking of which: the clean way to do this is have the inverse of `L.CRS.scale()`. I am assuming a scale of 2^zoom for this calculation, and I'm aware this might not be always true.